src/query/block/scalar.go

```diff
@@ -166,12 +174,15 @@ func (it *scalarSeriesIter) Next() bool {
 	return it.idx == 0
 }
 
-func (it *scalarSeriesIter) Current() (Series, error) {
+func (it *scalarSeriesIter) Current() Series {
 	if it.idx != 0 {
-		return Series{}, fmt.Errorf("invalid scalar index: %d, numVals: 1", it.idx)
+		// Indicates an error with the caller, having either not called Next() or attempted
```

Comment:
In most other places, we don't check if Next() has been called before Current() ? Also, having a panic here while you also have an Err() field is confusing. Maybe just get rid of this check ?

Reply:
Sure, just a lot of iterators work on top of slices so if you do something like call `Current` before `Next` or after `Next` is false, you'll get an out of bounds panic, so was to replicate that. Happy to delete it, it's not really a useful case, had it here so that it would be easier to detect if we were doing something wrong in the caller.